walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,108 +14,255 @@
 
 package com.google.sps;
 
+import java.util.Arrays;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.HashSet;
+import java.util.stream.Collectors;
 
 public final class FindMeetingQuery {
-    
+  
+  /** 
+   * First Solution: 
+   * Separate {@code events} in two lists based on attendee type.
+   * Call .findOpenTimeRanges() on each list
+   * Return a collection with valid time slots accroding to requesst.
+   *
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
     // If the request has no attendees, the open time slot is the entire day.
     if (request.getAttendees().isEmpty() && 
         request.getOptionalAttendees().isEmpty()) {
         return Arrays.asList(TimeRange.WHOLE_DAY);
     }
 
     // Sort {@code events} by starting time.
-    List<Event> eventsList = new ArrayList(events);
-    eventsList.sort((Event a, Event b) -> {
-        return TimeRange.ORDER_BY_START.compare(a.getWhen(), b.getWhen());
-    });
+    List<Event> eventsList = new ArrayList<>(events);;
```

Comment:
I thought about changing findOpenTimeRanges to accepting TimeRange objects, but when I create the variables mandatoryEvents and optionalEvents, I need to differentiate them from each other by using their attendees which I can only access when they are Event objects.

Reply:
Oh okay I changed it. At first, I didn't want to change the way that findOpenTimeRanges worked since I had to differentiate between optional and mandatory attendees, but I just did that before I called the method and had to create another variable for all event times.